app/src/test/kotlin/io/element/android/app/KonsistTest.kt

```diff
@@ -0,0 +1,139 @@
+import com.lemonappdev.konsist.api.ext.list.parameters
+import com.lemonappdev.konsist.api.ext.list.properties
+import com.lemonappdev.konsist.api.ext.list.withAllAnnotationsOf
+import com.lemonappdev.konsist.api.ext.list.withAllParentsOf
+import com.lemonappdev.konsist.api.ext.list.withNameEndingWith
+import com.lemonappdev.konsist.api.ext.list.withReturnType
+import com.lemonappdev.konsist.api.ext.list.withTopLevel
+import com.lemonappdev.konsist.api.ext.list.withoutName
+import com.lemonappdev.konsist.api.ext.list.withoutNameEndingWith
+import com.lemonappdev.konsist.api.verify.assertFalse
+import com.lemonappdev.konsist.api.verify.assertTrue
+import io.element.android.libraries.architecture.Presenter
+import io.element.android.libraries.designsystem.preview.PreviewsDayNight
+import org.junit.Test
+
+class KonsistTest {
+
+    @Test
+    fun `Classes extending 'Presenter' should have 'Presenter' suffix`() {
+        Konsist.scopeFromProject()
+            .classes()
+            .withAllParentsOf(Presenter::class)
+            .assertTrue {
+                it.name.endsWith("Presenter")
+            }
+    }
+
+    @Test
+    fun `Functions with '@PreviewsDayNight' annotation should have 'Preview' suffix`() {
+        Konsist
+            .scopeFromProject()
+            .functions()
+            .withAllAnnotationsOf(PreviewsDayNight::class)
+            .assertTrue {
+                it.hasNameEndingWith("Preview") &&
+                    it.hasNameEndingWith("LightPreview").not() &&
+                    it.hasNameEndingWith("DarkPreview").not()
+            }
+    }
+
+    @Test
+    fun `Top level function with '@Composable' annotation starting with a upper case should be placed in a file with the same name`() {
+        Konsist
+            .scopeFromProject()
+            .functions()
+            .withTopLevel()
+            .withoutModifier(KoModifier.PRIVATE)
+            .withoutNameEndingWith("Preview")
+            .withAllAnnotationsOf(Composable::class)
+            .withoutName(
+                // Add some exceptions...
+                "OutlinedButton",
+                "TextButton",
+                "SimpleAlertDialogContent",
+            )
+            .assertTrue(
+                additionalMessage =
+                """
+                Please check the filename. It should match the top level Composable function. If the filename is correct:
+                - consider making the Composable private or moving it to its own file
+                - at last resort, you can add an exception in the Konsist test
+                """.trimIndent()
+            ) {
+                if (it.name.first().isLowerCase()) {
+                    true
+                } else {
+                    val fileName = it.containingFile.name.removeSuffix(".kt")
+                    fileName == it.name
+                }
+            }
+    }
+
+    @Test
+    fun `Data class state MUST not have default value`() {
```

Comment:
This idea is to help the developer remember (by not compiling the code otherwise) that all the fields in the state should be provided when created by the presenter. So I think it's safer to enforce this for all the params.